addon/components/domless-glimmer.ts

```diff
@@ -9,21 +9,15 @@ const WILL_DESTROY = Symbol('will_destroy');
 const DESTROYING = Symbol('destroying');
 const DESTROYED = Symbol('destroyed');
 
-let ARGS_SET: WeakSet<any>;
-
-if (DEBUG) {
-  ARGS_SET = new WeakSet();
-}
-
-export { DESTROYING, DESTROYED, ARGS_SET };
+export { DESTROYING, DESTROYED };
 
 export interface DomlessGlimmerArgs<C> {
   parent?: DomlessGlimmerComponent<C, DomlessGlimmerArgs<C>>;
 }
 
 export default class DomlessGlimmerComponent<C = object, T extends DomlessGlimmerArgs<C> = object> {
   constructor(owner: unknown, args: T) {
-    if (DEBUG && !(owner !== null && typeof owner === 'object' && ARGS_SET.has(args))) {
```

Comment:
The previous code here (that we copied from Glimmer I guess?) didn't allow to pass different args to super() that the child component received. But that was only a dev-mode check, so seems not to be inherently required. Don't really know what the rationale was? 🤔

Reply:
This was copied from the base glimmer component indeed. I remember seeing a similar question on the need for it in discord but haven't seen the answer. I think we can safely remove it for our purposes though.